src/qarchivecompressor_p.cc

```diff
@@ -229,69 +229,31 @@ void CompressorPrivate::addFiles(const QStringList &entryNames, const QStringLis
 }
 
 void CompressorPrivate::removeFiles(const QString &entry) {
-    if(b_Started || b_Paused) {
+    if(b_Started || b_Paused)
         return;
-    }
-    int index = 0;
-    for(const auto& file : m_StaggedFiles) {
-        if(file && file->entry == entry) {
-            m_StaggedFiles.remove(index);
-            return;
-        }
-        ++index;
-    }
+
+    std::for_each(entries.begin(), entries.end(), removeFile);
 }
 
 void CompressorPrivate::removeFiles(const QStringList &entries) {
```

Comment:
This should be the single entry remove, so should take QString..

Reply:
or just revert your commit and change QStringList overloaded method correctly

Reply:
Done.